src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitTest.java

```diff
@@ -0,0 +1,50 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.methods;
+
+import org.junit.jupiter.api.Test;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.lang.ArchRule;
+
+public class ArchUnitTest {
+
+    /**
+     * Test contains asserts in callstack, but idea does not see them.
+     *
+     * @noinspection JUnitTestMethodWithNoAssertions
+     */
+    @Test
+    public void nonProtectedCheckMethodsTest() {
+        final JavaClasses importedClasses = new ClassFileImporter()
+                .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_TESTS)
+                .importPackages("com.puppycrawl.tools.checkstyle.checks");
+
+        final ArchRule checkMethodsShouldNotBeProtectedRule = methods().that()
+            .areDeclaredInClassesThat().haveNameMatching("^(Abstract).*Check")
```

Comment:
https://github.com/checkstyle/checkstyle/issues/7469#issuecomment-597433647
> Checks should not have protected methods or fields.

By the name matching `"^(Abstract)`, isn't this only checking that only methods in classes named "Abstract" not have protected methods? If so

Reply:
good catch.

> ^(Abstract)

should be changed to all Checks except for `Abstract...Check`

Reply:
@romani @gaurabdg If you agree with my analysis that we are not checking the right class names, why isn't there a violation in travis?

If it is scanning abstract classes, why didn't it flag them as I pointed out we have non-override protected methods in an abstract class. If it is not scanning abstract classes, why didn't it flag the methods with the Override annotations. Nothing says to specifically ignore overridden methods and it seems always skipping or defaulting to skip them with lead to issues with people unfamilar with arch unit.

I couldn't confirm anything in their documentation and didn't try to debug this more.

Reply:
While fixing this, I tried out two tests:
```
        final ArchRule checkMethodsShouldNotBeProtectedRule =
            methods().that()
            .areNotAnnotatedWith("Override").and()
            .areDeclaredInClassesThat()
            .doNotHaveModifier(JavaModifier.ABSTRACT)
            .should().notBeProtected();
```

```
final ArchRule checkMethodsShouldNotBeProtectedRule =
    methods().that()
    .areProtected().and()
    .areDeclaredInClassesThat()
    .doNotHaveModifier(JavaModifier.ABSTRACT)
    .should().beAnnotatedWith("Override");
```

Both gave the same violations (92 of them). I'll ask TNG regarding this and get back to you.
```
java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'methods that are not annotated with @Override and are declared in classes that do not have modifier ABSTRACT should not be protected' was violated (92 times):
Method <com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (NewlineAtEndOfFileCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.OrderedPropertiesCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (OrderedPropertiesCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.TranslationCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (TranslationCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (UniquePropertiesCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$AnonymousClassFrame.getFrameName()> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$BlockFrame.getType()> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.containsFieldOrVariable(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.getIfContains(com.puppycrawl.tools.checkstyle.api.DetailAST, boolean)> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.getType()> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.isProperDefinition(com.puppycrawl.tools.checkstyle.api.DetailAST, com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ConstructorFrame.getType()> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$MethodFrame.getType()> has modifier PROTECTED in (RequireThisCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.SuperCloneCheck.getMethodName()> has modifier PROTECTED in (SuperCloneCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.coding.SuperFinalizeCheck.getMethodName()> has modifier PROTECTED in (SuperFinalizeCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.header.HeaderCheck.postProcessHeaderLines()> has modifier PROTECTED in (HeaderCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.header.HeaderCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (HeaderCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.header.RegexpHeaderCheck.postProcessHeaderLines()> has modifier PROTECTED in (RegexpHeaderCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.header.RegexpHeaderCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (RegexpHeaderCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.imports.FileImportControl.matchesExactly(java.lang.String, java.lang.String)> has modifier PROTECTED in (FileImportControl.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.imports.PkgImportControl.matchesExactly(java.lang.String, java.lang.String)> has modifier PROTECTED in (PkgImportControl.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.canChildrenBeNested()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.curlyIndent()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.getChildrenExpectedIndent()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.getIndentImpl()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.getLeftCurly()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.getListChild()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.getRightCurly()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ArrayInitHandler.getTopLevelAst()> has modifier PROTECTED in (ArrayInitHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.canChildrenBeNested()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.checkTopLevelToken()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.curlyIndent()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.getCheckedChildren()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.getChildrenExpectedIndent()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.getLeftCurly()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.getListChild()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.getNonListChild()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.getRightCurly()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.BlockParentHandler.getTopLevelAst()> has modifier PROTECTED in (BlockParentHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.CaseHandler.getIndentImpl()> has modifier PROTECTED in (CaseHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ClassDefHandler.getCheckedChildren()> has modifier PROTECTED in (ClassDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ClassDefHandler.getLeftCurly()> has modifier PROTECTED in (ClassDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ClassDefHandler.getListChild()> has modifier PROTECTED in (ClassDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ClassDefHandler.getRightCurly()> has modifier PROTECTED in (ClassDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ClassDefHandler.getTopLevelAst()> has modifier PROTECTED in (ClassDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ElseHandler.checkTopLevelToken()> has modifier PROTECTED in (ElseHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ElseHandler.getNonListChild()> has modifier PROTECTED in (ElseHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.IfHandler.checkTopLevelToken()> has modifier PROTECTED in (IfHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.IfHandler.getIndentImpl()> has modifier PROTECTED in (IfHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.LabelHandler.getIndentImpl()> has modifier PROTECTED in (LabelHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.LambdaHandler.getIndentImpl()> has modifier PROTECTED in (LambdaHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.MemberDefHandler.checkModifiers()> has modifier PROTECTED in (MemberDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.MethodCallHandler.getIndentImpl()> has modifier PROTECTED in (MethodCallHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.MethodCallHandler.shouldIncreaseIndent()> has modifier PROTECTED in (MethodCallHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.MethodDefHandler.checkModifiers()> has modifier PROTECTED in (MethodDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.MethodDefHandler.getTopLevelAst()> has modifier PROTECTED in (MethodDefHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.NewHandler.getIndentImpl()> has modifier PROTECTED in (NewHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.NewHandler.shouldIncreaseIndent()> has modifier PROTECTED in (NewHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ObjectBlockHandler.curlyIndent()> has modifier PROTECTED in (ObjectBlockHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ObjectBlockHandler.getIndentImpl()> has modifier PROTECTED in (ObjectBlockHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ObjectBlockHandler.getLeftCurly()> has modifier PROTECTED in (ObjectBlockHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ObjectBlockHandler.getListChild()> has modifier PROTECTED in (ObjectBlockHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ObjectBlockHandler.getRightCurly()> has modifier PROTECTED in (ObjectBlockHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.ObjectBlockHandler.getTopLevelAst()> has modifier PROTECTED in (ObjectBlockHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.PrimordialHandler.getIndentImpl()> has modifier PROTECTED in (PrimordialHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SlistHandler.getLeftCurly()> has modifier PROTECTED in (SlistHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SlistHandler.getListChild()> has modifier PROTECTED in (SlistHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SlistHandler.getRightCurly()> has modifier PROTECTED in (SlistHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SlistHandler.getTopLevelAst()> has modifier PROTECTED in (SlistHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SwitchHandler.getLeftCurly()> has modifier PROTECTED in (SwitchHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SwitchHandler.getListChild()> has modifier PROTECTED in (SwitchHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SwitchHandler.getNonListChild()> has modifier PROTECTED in (SwitchHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.indentation.SwitchHandler.getRightCurly()> has modifier PROTECTED in (SwitchHandler.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocPackageCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (JavadocPackageCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.metrics.ClassDataAbstractionCouplingCheck.getLogMessageId()> has modifier PROTECTED in (ClassDataAbstractionCouplingCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.metrics.ClassFanOutComplexityCheck.getLogMessageId()> has modifier PROTECTED in (ClassFanOutComplexityCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.CatchParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (CatchParameterNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.ClassTypeParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (ClassTypeParameterNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (ConstantNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.InterfaceTypeParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (InterfaceTypeParameterNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.LambdaParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (LambdaParameterNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.LocalFinalVariableNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (LocalFinalVariableNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.LocalVariableNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (LocalVariableNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.MemberNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (MemberNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.MethodTypeParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (MethodTypeParameterNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.ParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (ParameterNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.naming.StaticVariableNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> has modifier PROTECTED in (StaticVariableNameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpMultilineCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (RegexpMultilineCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpOnFilenameCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (RegexpOnFilenameCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (RegexpSinglelineCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.sizes.FileLengthCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (FileLengthCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (LineLengthCheck.java:0)
Method <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck.processFiltered(java.io.File, com.puppycrawl.tools.checkstyle.api.FileText)> has modifier PROTECTED in (FileTabCharacterCheck.java:0)
```

Reply:
I wrote this condition for the test:
```
        final ArchRule checkMethodsShouldNotBeProtectedRule =
            methods().that()
            .areNotAnnotatedWith(Override.class).and()
            .areDeclaredInClassesThat()
            .haveSimpleNameEndingWith("Check").and()
            .doNotHaveModifier(JavaModifier.ABSTRACT)
            .should().notBeProtected();
```

The error reported was as follows:
`com.tngtech.archunit.base.ArchUnitException$InvalidSyntaxUsageException: Annotation type java.lang.Override has @Retention(SOURCE), thus the information is gone after compile. So checking this with ArchUnit is useless.
`
So, we can't essentially use the `Override` annotation to skip methods such as `processFiltered`. I am looking at other workarounds.

Reply:
One thing I can do is to collect all `Abstract.*` class' methods and generate a regex string like `.*(processFiltered | getMethodName...)` to add to the archrule condition `..methods().that().haveNameNotMatching(regex)`. Shall I implement this?

Reply:
> One thing I can do is to collect all Abstract.* class' methods

Can ArchUnit do this for you so you don't have to? Or is it strictly a violation tool.

I don't see any other choice for specific methods unless we create some new annotation to line up with `Override`. I would add some type of comment that says what the method list is.

What about my concern for the currently pushed code not producing a violation when it looks like it should be?

Reply:
> What about my concern for the currently pushed code not producing a violation when it looks like it should be?

I was assuming it was ignoring only `Abstract.*` checks, but it skipped all, it was a mistake on my part.

> Can ArchUnit do this for you so you don't have to? Or is it strictly a violation tool.

I will look into it.

> I don't see any other choice for specific methods unless we create some new annotation to line up with `Override`.

Yes, we can create a new annotation with custom retention.

> I would add some type of comment that says what the method list is.

I didn' understand what you are trying to say. Are you saying we should create an `ignoreList` set, to ignore?

Reply:
> I didn' understand what you are trying to say.

If we create a list of methods to suppress, we need a comment saying something like "these methods are a list of methods with the Override annotation". This way it is clear when new Override method is added, we know it should be added to the list and should not add a method that is protected without an Override.

@romani please chime in on your thoughts on workaround. It seems ArchUnit has some limitations.

Reply:
> as the whole point of abstract classes is to have protected methods.

is to have same implementation of some methods irrespective of modifier. Projected methods are visible not only Child classes, BUT to same package classes also - this is the biggest problems of them, they have two sides of visibility (NOT a single).

Reply:
@gaurabdg ,

> So, we can't essentially use the Override annotation to skip methods such as processFiltered. I am looking at other workarounds.

please report this to ArchUnit maintainers and ask them for for workaround, there might be some option to disable this error. Static analysys tool need to work on annotations that have no effect in runtime, majority of annotations are flags/metadata/markers.

Reply:
Here is the issue https://github.com/TNG/ArchUnit/issues/359
If they don't reply within a few days, I'll implement the workaround.

Reply:
Here is the reply https://github.com/TNG/ArchUnit/issues/359#issuecomment-631054149, its not possible it seems.

Reply:
rnveach od not have time to continue this PR, I take over it.